File: WoofIOS/SwiftPackages/Presentation/Profile/Tests/PresentationProfileTests/Factory/DogFactory.swift
import Models

final class DogFactory {
    private var dog = Dog(id: 1, name: "Bruno", photoUrl: "woof://dogs/golden_retriever_01.jpg")

    @discardableResult
    func withId(_ id: Int) -> Self {
        dog = Dog(id: id, name: "Dog \(id)", photoUrl: dog.photoUrl)
        return self
    }

    func produce() -> Dog {
        dog
    }
}

File: WoofAndroid/presentation/profile/src/test/kotlin/com/perrystreet/woof/presentation/profile/factory/DogFactory.kt
package com.perrystreet.woof.presentation.profile.factory

import com.perrystreet.woof.models.dog.Dog

class DogFactory {
    private var dog = Dog(id = 1, name = "Bruno", photoUrl = "file:///android_asset/dogs/golden_retriever_01.jpg")

    fun withId(id: Long) = apply {
        dog = dog.copy(id = id, name = "Dog $id")
    }

    fun produce(): Dog = dog
}
